src/nbl/video/COpenGLCommandBuffer.h

```diff
@@ -567,6 +572,9 @@ class COpenGLCommandBuffer final : public IGPUCommandBuffer
             if (!this->isCompatibleDevicewise(buffer))
                 return false;
 
+        if (!emplace<COpenGLCommandPool::CBindIndexBufferCmd>(core::smart_refctd_ptr<const buffer_t>(buffer), offset, indexType))
+            return false;
+
```

Comment:
if you accept my QoL idea, then this would change into a 
```cpp
bool bindIndexBuffer(const buffer_t* buffer, size_t offset, asset::E_INDEX_TYPE indexType) override
{
    // all non-nullity and compatibility checks hoisted to `IGPUCommandBuffer`
    auto* cmd = IGPUCommandBuffer::bindIndexBuffer_impl(core::smart_refctd_ptr<const buffer_t>(buffer));
    if (!cmd) // validation of arguments or allocation failed
        return false;

    // we pass the pointer to the generic command, because if backend-specific fails to allocate, we need to cancel the specific (also we could store a pointer to the generic in the specific to get some funky data)
    return getGLCommandPool()->emplace<COpenGLCommandPool::CBindIndexBufferCmd>(cmd,offset,indexType));
} 
```

Reply:
Resolved.